### PR TITLE
feat: SRS-552 - mapSearch query

### PR DESCRIPTION
**Pull request by anton-bcgov**
# Description

This PR add a separate `mapSearch` query to be used in Map Search. Previously this page reused `siteSearch` query.

Fixes # ([SRS-552](https://env-epd.atlassian.net/browse/SRS-552))

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Navigate to `/map` route, validate that `mapSearch` query is being run there

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Backend](https://nr-site-registry-135-backend.apps.silver.devops.gov.bc.ca)
- [Frontend](https://nr-site-registry-135-frontend.apps.silver.devops.gov.bc.ca)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-site-registry/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-site-registry/actions/workflows/merge.yml)

### CODE REVIEW

File: backend/src/app/services/site/site.service.ts [hunk 1]
```diff
@@ -265,6 +266,45 @@ export class SiteService {
     return response;
   }
 
+  async mapSearch(searchTerm = '') {
```
Comment (anton-bcgov) [inline backend/src/app/services/site/site.service.ts:270]: I plopped this into Site service, but if anyone has strong objections, I can also move it into its own separate service. Let me know what you think!

File: backend/src/app/dto/response/genericResponse.ts [hunk 2]
```diff
@@ -21,6 +21,22 @@ export class FetchSiteResponse extends BaseHttpResponse {
   }
 }
 
+@ObjectType()
+export class MapSearchResponse extends BaseHttpResponse {
```
Comment (jaspalsingh-aot) [inline backend/src/app/dto/response/genericResponse.ts]: can we you generic response in order to make consistency in code?

File: backend/src/app/services/site/site.service.ts (hunk 1 above)
Comment (bolny) [reply backend/src/app/services/site/site.service.ts:270]: I think searching for sites belongs in the site service personally.

**Comment by acoard-aot** [comment]
Closing PR, so we can re-open #143  as we're limited by storage in OpenShift. Feel free to re-open this in 20minutes if I forgot.

**Comment by acoard-aot** [comment]
Re-opening after #143 has successfully deployed. Note: The action for Deploy WILL FAIL currently, that is to be expected, as #143 is consuming all the storage quota.  We will hopefully have this resolved soon once we have a new temporary environment provisioned for us.

File: backend/src/app/resolvers/site/site.resolver.ts [hunk 3]
```diff
@@ -329,4 +333,36 @@ export class SiteResolver {
...
+    try {
+      const data = await this.siteService.mapSearch(searchParam);
+      return this.mapSearchGenericResponseProvider.createResponse(
+        'Successfully fetched sites for map',
+        HttpStatus.OK,
+        true,
+        data,
+      );
+    } catch (e) {
+      return this.mapSearchGenericResponseProvider.createResponse(
```
Comment (jaspalsingh-aot) [inline backend/src/app/resolvers/site/site.resolver.ts:240]: do we need logger for catch block ?

**Review by jaspalsingh-aot — CHANGES_REQUESTED** [review]

**Review by jaspalsingh-aot — APPROVED** [review]
Seems fine to me

**Pull request merged**